resources/fetchersManager/data.go

```diff
@@ -19,27 +19,34 @@ package fetchersManager
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"sync"
 	"time"
 
-	"github.com/elastic/beats/v7/libbeat/common/atomic"
 	"github.com/elastic/cloudbeat/resources/fetching"
 	"github.com/elastic/elastic-agent-libs/logp"
+	"go.uber.org/atomic"
 )
 
 // Data maintains a cache that is updated by Fetcher implementations registered
 // against it. It sends the cache to an output channel at the defined interval.
 type Data struct {
-	log        *logp.Logger
-	timeout    time.Duration
-	interval   time.Duration
-	fetchers   FetchersRegistry
-	stop       chan struct{}
+	log *logp.Logger
+
+	// Maximum duration a fetcher is allowed to run
+	timeout time.Duration
+
+	// Interval between two consecutive cycles
+	interval time.Duration
+
+	fetchers FetchersRegistry
+
+	// Maximum timeout for graceful shutdown
 	stopNotice chan time.Duration
 }
 
-type Stop func(context.Context, time.Duration)
```

Comment:
`Stop` creates its own `WithCancel` context, derived from the `bt.ctx` passed to `Run()`, which is the same context we used to pass here.